### PR TITLE
Ramad personal page

### PR DESCRIPTION
**Pull request by michael-hoffman-26**
start working on the ramad personal page.
1 did fake fetch for the ramad private ads,
 update the display of ads to work with Type Objects.

2 activate the WITH_RAMAD_ACCESS variable to detrmine isRamad

### CODE REVIEW

File: server/routes/auth/auth.js [hunk 1]
```diff
@@ -32,17 +34,28 @@ router.get('/auth/callback',
...
 
-    /* todo detrmine isRamad acording to WITH_RAMAD_ACCESS from '.env'
-    giving priority to the .env variable, need to convert string to boolean to*/
+    const isRamad = checkIsRamad(WITH_RAMAD_ACCESS, user.is_ramad);
     
     res.cookie('user', JSON.stringify({
         upn: user.upn,
         name: user.name,
-        isRamad: user.is_ramad,
+        isRamad: isRamad,
```
Comment (JetoPistola) [inline server/routes/auth/auth.js:42]: `isRamad` , as keyAndValue: keyAndValue, to keyAndValue 😃

File: server/routes/auth/auth.js [hunk 2]
```diff
@@ -32,17 +34,28 @@ router.get('/auth/callback',
...
 
     // redirect home
     res.redirect('/');
   });
 
-module.exports = router;
\ No newline at end of file
+module.exports = router;
+
+const checkIsRamad = (envIsRamad, userIsRamad) => {
```
Comment (JetoPistola) [inline server/routes/auth/auth.js:51]: 💯 Super!!!

File: src/components/Personal-Zone/Publisher/JobsAppEmployerBody.tsx [hunk 3]
```diff
@@ -21,27 +21,27 @@ const styles = makeStyles({
...
+    const userStore: UserStore = props.userStore!;
 
     useEffect(() => {
-        jobsStore.loadAdvertisements();
-    }, [jobsStore]);
+        userStore.loadRamadAds();
+    }, [userStore]);
 
-    const showEmptyDisplay = false && jobsStore.advertisements.length > 0; 
+    const showEmptyDisplay = userStore.getRamadAds.length === 0
```
Comment (JetoPistola) [inline src/components/Personal-Zone/Publisher/JobsAppEmployerBody.tsx:35]: 💯

File: src/components/Personal-Zone/Publisher/JobsAppEmployerBody.tsx [hunk 4]
```diff
@@ -21,27 +21,27 @@ const styles = makeStyles({
...
     return (
         <div className={classes.appBodyContent}>
-            <Header withAddButton={showEmptyDisplay}/>
+            <Header withAddButton={!showEmptyDisplay}/> {/*There are ads, show the addButton*/}
```
Comment (JetoPistola) [inline src/components/Personal-Zone/Publisher/JobsAppEmployerBody.tsx:38]: 💯

File: src/server/user.ts [hunk 5]
```diff
@@ -23,10 +25,47 @@ export async function setFavoriteAd(adId: number): Promise<boolean> {
...
-        console.log(response)
         // Need to think how to determine if the action succeeded
         return response.status === 200;
     });
 
     return answer;
+};
+
+export async function fetchRamadAds(): Promise<RamadAds[]> {
+    const ramadAds: RamadAds[] = [
```
Comment (JetoPistola) [inline src/server/user.ts:36]: // TODO: insert DDL

File: src/types/User.ts [hunk 6]
```diff
@@ -7,6 +8,18 @@ export interface User {
     isRamad: boolean;
 }
 
+export interface RamadAds {
+    id: number;
+    name: string;
+    role: RoleAdPropertiesData;
+    candidate: Condidate[];
```
Comment (JetoPistola) [inline src/types/User.ts:15]: `candidates: Candidate[]`

1. plural (Could be 2+ candidates, that's why the 's' in the end 😀)
2. Typo, `Candidate` and not `Condidate`

File: server/routes/auth/auth.js [hunk 7]
```diff
@@ -2,6 +2,8 @@ const express = require('express');
 const router  = express.Router();
 const passport = require('passport');
 const db = require('../../db');
+require('dotenv').config();
```
Comment (stavbi1) [inline server/routes/auth/auth.js:5]: Redundant, you only need to conigure once, and we did it in index.js

File: server/routes/auth/auth.js (hunk 2 above)
Comment (stavbi1) [reply server/routes/auth/auth.js:51]: add return type

File: src/components/Personal-Zone/Publisher/JobsAppEmployerBody.tsx (hunk 3 above)
Comment (stavbi1) [reply src/components/Personal-Zone/Publisher/JobsAppEmployerBody.tsx:35]: add semicolon

**Comment by stavbi1** [comment]
Approved

File: server/routes/auth/auth.js (hunk 2 above)
Comment (michael-hoffman-26) [reply server/routes/auth/auth.js:51]: that is a `js` file not `ts`

File: server/routes/auth/auth.js (hunk 7 above)
Comment (michael-hoffman-26) [reply server/routes/auth/auth.js:5]: OK good to know, thanks, I removed it

File: src/components/Personal-Zone/Publisher/JobsAppEmployerBody.tsx (hunk 3 above)
Comment (michael-hoffman-26) [reply src/components/Personal-Zone/Publisher/JobsAppEmployerBody.tsx:35]: that is a js file not ts, added semicolon

File: src/server/user.ts (hunk 5 above)
Comment (michael-hoffman-26) [reply src/server/user.ts:36]: Done

File: src/types/User.ts (hunk 6 above)
Comment (michael-hoffman-26) [reply src/types/User.ts:15]: Thanks done

File: server/routes/auth/auth.js (hunk 1 above)
Comment (michael-hoffman-26) [reply server/routes/auth/auth.js:42]: Done

**Pull request merged**